exploration/syntax-exploration-2.md

```diff
@@ -0,0 +1,256 @@
+- Uses a sigil for code statements. 
+  - Requires the code-mode sigil to be escaped in **_unquoted_** patterns.
+  - Quoted patterns do not require this sigil to be escaped.
+  - Note that decisions about auto-trimming affect the escaping situation. 
+    If quoting the pattern is required, no additional escapes are needed.
+- Placeholders and expressions use `{`/`}`:
+  > `{$var}`, `{unquoted}`, `{|quoted|}`, `{$var :function}`
+
+The use of `%` as the sigil is **_not_** an inherent part of the design here.
+The actual sigil or sigil sequence needs to be decided. 
+Note that using a doubled-sigil reduces the need for escaping.
+
+Patterns in this syntax are generally unquoted,
+but MAY be quoted with the 2023-10-09 consensus `{{`/`}}` quotes.
+
+Patterns lacking declarations, selectors, or both, are **_not_** auto-trimmed.
+Whitespace in quoted patterns is significant.
+Whitespace in unquoted patterns may or may not be significant (hot topic).
+
+```
+Hello world!
+```
+```
+Hello {$user}!
+```
+```
+%input {$var :function option=value}
+Hello {$var}
+
+%input {$var :function option=value}Hello {$var}
+```
+```
+%input {$var :function option=value}
+%local $foo = {$bar :function option=value}
+Hello {$var}, you have a {$foo}
+
+%input {$var :function option=value}%local $foo = {$bar :function option=value}Hello {$var}, you have a {$foo}
+```
+```
+%match {$foo}
+%when{foo} Hello {$foo} you have a {$var}
+%when{*} {$foo} hello you have a {$var}
+
+%match {$foo}%when{foo}Hello {$foo} you have a {$var}%when{*}{$foo} hello you have a {$var}
+```
+
+```
+%match {$foo :function option=value} {$bar :function option=value}
+%when {a b} {{  {$foo} is {$bar}  }}
+%when {x y} {||}  {$foo} is {$bar}  {||}
+%when {* *} {|  |}{$foo} is {$bar}{|  |}
+
+%match {$foo :function option=value}{$bar :function option=value}%when {a b}{{  {$foo} is {$bar}  }}%when {x y} {||}{$foo} is {$bar}  {||}%when {* *}{|  |}{$foo} is {$bar}{|  |}
+```
+
+## Whitespace and Pattern Boundaries
+
+One key consideration for the above options is distinguishing the boundary between 
+the end of a _pattern_ and the start of a _variant_ (`when` clause).
+This was not an issue in the code-mode syntax because the patterns were always delimited ("quoted").
+
+> [*Note]
+> This type of syntax is only needed if patterns are not quoted.
+> The "hot debate" in our 2023-10-16 teleconference was between "WYSIWYG"
+> no-whitespace-trimming supporters and current WG consensus.
+> For current consensus and discussion of whitespace design see [here](pattern-exterior-whitespace.md)
+
+- Candidate 1a uses a double sigil `{#` or distinguishing sequence `{#when` to accomplish this.
+- Candidate 2a quotes all patterns in code-mode.
+- Candidate 3a uses a sigil-keyword sequence `%when` that required at least some additional escaping.
+
+It is reasonable to think that we might modify this particular part of the syntax
+to improve usability. **_Keep in mind the need for single-line authoring._**
```

Comment:
Do "single-line authoring" mean that:
- you must be able to express any possible functionality on a single line
or
- Carriage Returns are irrelevant. Any MF2.0 string must render the same if all CRs are removed.
or
- something else?

Reply:
It's the former. Single-line authoring means that syntactically-allowed-but-not-required whitespace is removed. It is shorthand for: many people will author messages in a resource format in which the _message_ is a string with file-local escaping. Think about that in addition to the pretty formatting we use in examples like this:

```
{input $var :function opt=val}
{%match {$foo}}
{%when foo}
    You have {$foo}
{$when *}
    You have really {$foo}
```

That might be single line as:

```
myMessage = "{input $var :function opt=val}{%match {$foo}}{%when foo}You have {$foo}{$when *}You have really {$foo}"
```

(Note that I have trimmed whitespace off of the pattern in the single-line example.)